File: 6_kyu/Create_Phone_Number.go
package kata

import "strings"

func writeDigits(dst *strings.Builder, digits []uint) {
	for _, value := range digits {
		if value < 10 {
			dst.WriteRune(rune('0' + value))
		}
	}
}

func CreatePhoneNumber(numbers [10]uint) string {
	var dst strings.Builder
	dst.WriteRune('(')
	writeDigits(&dst, numbers[0:3])
	dst.WriteString(") ")
	writeDigits(&dst, numbers[3:6])
	dst.WriteRune('-')
	writeDigits(&dst, numbers[6:10])
	return dst.String()
}

File: 6_kyu/Create_Phone_Number.rs
fn create_phone_number(numbers: &[u8]) -> String {
    let digits: String = numbers.iter().map(|&i| (i + b'0') as char).collect();
    format!("({}) {}-{}", &digits[0..3], &digits[3..6], &digits[6..])
}
